packages/core/src/utilities/createFloat32SharedArray.ts

```diff
@@ -24,21 +24,26 @@ import global from '../global';
  * @returns a Float32Array with an underlying SharedArrayBuffer
  * @public
  */
-function createFloat32SharedArray(length: number): Float32Array {
-  if (!window.crossOriginIsolated) {
+function createFloat32SharedArray(
+  length: number,
+  useSharedArrayBuffer = true
+): Float32Array {
+  if (!window.crossOriginIsolated && useSharedArrayBuffer) {
     throw new Error(
       'Your page is NOT cross-origin isolated, see https://developer.mozilla.org/en-US/docs/Web/API/crossOriginIsolated'
     );
   }
-  if (window.SharedArrayBuffer === undefined) {
+  if (window.SharedArrayBuffer === undefined && useSharedArrayBuffer) {
     throw new Error(
       'SharedArrayBuffer is NOT supported in your browser see https://developer.chrome.com/blog/enabling-shared-array-buffer/'
     );
   }
 
-  const sharedArrayBuffer = new SharedArrayBuffer(length * 4);
+  const buffer = useSharedArrayBuffer
+    ? new SharedArrayBuffer(length * 4)
+    : new ArrayBuffer(length * 4);
 
-  return new Float32Array(sharedArrayBuffer);
+  return new Float32Array(buffer);
```

Comment:
I don't think you should do it here. This file should do what it says (creating sharedArray). Maybe in the cornerstoneStreamingImageVolume you branch to use non-shared